aten/src/ATen/native/LinearAlgebra.cpp

```diff
@@ -1322,6 +1322,46 @@ Tensor outer(const Tensor& self, const Tensor& vec2) {
             TYPE, NAME, __VA_ARGS__)
 #endif
 
+
+#ifdef __aarch64__
+static inline int64_t get_mkldnn_matmul_min_dim() {
+  static auto value = [&] {
+    // Minimum dimension requirement for MKLDNN; derived based on experiments.
+    constexpr int64_t default_min_dim = 8;
+    const char* ptr = std::getenv("TORCH_MKLDNN_MATMUL_MIN_DIM");
+    return ptr != nullptr ? std::atoi(ptr) : default_min_dim;
+  }();
+  return value;
+}
+
+static inline int64_t get_mkldnn_matmul_min_size() {
+  static auto value = [&] {
+    // Minimum size requirement for MKLDNN; derived based on experiments.
+    constexpr int64_t default_min_size = 8 * 1024;
+    const char* ptr = std::getenv("TORCH_MKLDNN_MATMUL_MIN_SIZE");
+    return ptr != nullptr ? std::atoi(ptr) : default_min_size;
+  }();
+  return value;
+}
+
+static inline bool apply_mkldnn_matmul_heur(int64_t m, int64_t k, int64_t n, int64_t min_dim, int64_t min_size) {
+  return m > min_dim && k > min_dim && n > min_dim && m * k * n > min_size;
```

Comment:
Q: why explicitly pass `min_dim`/`min_size` arguments when they can only be by calling the respective functions, which also have no other use. 
```suggestion
static inline bool apply_mkldnn_matmul_heur(int64_t m, int64_t k, int64_t n) {
  const auto min_dim = get_mkldnn_matmul_min_dim();
  const auto min_size = get_mkldnn_matmul_min_size();
  return m > min_dim && k > min_dim && n > min_dim && m * k * n > min_size;
```

Reply:
@malfet I changed the function signature based on your feedback.